client/service/src/lib.rs

```diff
@@ -125,7 +125,7 @@ where
 
 	task_manager
 		.spawn_essential_handle()
-		.spawn("cumulus-consensus", None, consensus);
+		.spawn_blocking("cumulus-consensus", None, consensus);
```

Comment:
Why is this is now classified as blocking? 

[here](https://github.com/paritytech/substrate/blob/74b2c92066ec3abcb612faa9272f246ae339fab3/client/service/src/config.rs#L153-L158) I can read that `TaskType::Blocking` is generally used when the task may block somehow the execution for pecause of expensive CPU ops or `sleep`s.

What are the synchronous ops that may "block" the executor here?

Reply:
This task receives finality notifications from the relay chain and finalizes accordingly.
Therefore it calls `finalize_block` on the backend [here](https://github.com/paritytech/cumulus/blob/master/client/consensus/common/src/parachain_consensus.rs#L71), which does potentially expensive tree-route calculations and db access. I prefer to let the TaskManager spawn a blocking task for this.

Reply:
Got it! Indeed traversing the chain can be quite expensive. Maybe a day we can make this `tree_route` async by spawning a dedicated task and waiting for the result on a channel (I already saw this pattern around the project IIRC)